ExomeDepth/make_BEDdetail.py

```diff
@@ -0,0 +1,365 @@
+                            stop=stop,
+                            calltype=calltype,
+                            ntargets=ntargets
+                            )
+                        if event not in event_dic:
+                            event_dic[event] = {"parent":{"count":0, "bf":[],"ratio":[],"correlation":[],"deldupratio":[],"totalcalls":[], "gender":[]}, \
+                                            "child":{"count":0, "bf":[],"ratio":[],"correlation":[],"deldupratio":[],"totalcalls":[], "gender":[]}}
+
+                        """ Determine Child or Parent status based on sampleid. There is no other option at the moment"""
+                        if "CM" in sampleid or "CF" in sampleid or "CO" in sampleid:
+                            sampletype = "child"
+                            childs +=1
+                        elif "PM" in sampleid or "PF" in sampleid or "PO" in sampleid:
+                            sampletype = "parent"
+                            parents += 1
+               
+                        event_dic[event][sampletype]["count"] += 1
+                        event_dic[event][sampletype]["bf"].append(bf)
+                        event_dic[event][sampletype]["ratio"].append(ratio)
+                        event_dic[event][sampletype]["correlation"].append(correl)
+                        event_dic[event][sampletype]["deldupratio"].append(deldupratio)
+                        event_dic[event][sampletype]["totalcalls"].append(totalcalls)
+                        event_dic[event][sampletype]["gender"].append(gender)  #  Is not being used in the BED file output at the moment.
+    all_event_file.close()
+    excluded_samples_file.close()
+    return event_dic, childs, parents
+
+def make_beddetail(args, event_dic, childs, parents):
+    event_file = open("{outputfolder}/{outputfile}.bed".format(outputfolder=args.outputfolder, outputfile=args.outputfile),"w")
+    event_file_igv = open("{outputfolder}/{outputfile}_IGV.bed".format(outputfolder=args.outputfolder, outputfile=args.outputfile),"w")
+
+    """ print header in BED file """
+    event_file.write("track name=\"HC_WES_CNV\" type=\"bedDetail\" description=\"CNVs called by Exomedepth using HC callset. #Child={childs} #Partent={parents} \" visibility=3 itemRgb=\"On\"\n".format(childs=childs, parents=parents)) 
+    event_file_igv.write("track name=\"HC_WES_CNV\" type=\"bed\" description=\"CNVs called by Exomedepth using HC callset. #Child={childs} #Partent={parents} \" visibility=3 itemRgb=\"On\"\n".format(childs=childs, parents=parents))
+    total_event_list = []
+    for item in event_dic:
+        chrom, start, stop, calltype, ntargets = item.split("_")
+        """ Make start 0-based for BED file """
+        start = int(start) - 1
+        event_list = []
+
+        if chrom == "X":
+            event_list.append(23)
+        elif chrom == "Y":
+            event_list.append(24)
+        else:
+            event_list.append(int(chrom))
```

Comment:
Waarom X, Y apart? En wat doe je met MT (die hebben we zeker niet)?

Als het om sorteren gaat, volgens mij werkt dat gewoon in python.